test/services/tagged_content_test.rb

```diff
@@ -0,0 +1,67 @@
+require 'test_helper'
+require './test/support/custom_assertions.rb'
+
+describe TaggedContent do
+  describe '#fetch' do
+    it 'returns the documents from search' do
+      search_results = mock(documents: ['doc1'])
+
+      RummagerSearch.stubs(:new).returns(search_results)
+
+      assert(tagged_content.fetch, ['doc1'])
+    end
+
+    it 'starts from the first page' do
+      assert_includes_params(start: 0)
+    end
+
+    it 'requests all results' do
+      assert_includes_params(count: RummagerSearch::PAGE_SIZE_TO_GET_EVERYTHING)
+    end
+
+    it 'requests a limited number of fields' do
+      assert_includes_params(fields: %w(title description link))
+    end
+
+    it 'orders the results by title' do
+      assert_includes_params(order: 'title')
+    end
+
+    it 'filters the results by taxon' do
+      assert_includes_params(filter_taxons: [taxon_content_id])
+    end
+
+    it 'filters out non-guidance content' do
+      guidance_document_types = RummagerSearch::GUIDANCE_DOCUMENT_TYPES
+
+      assert_includes_params(
+        filter_content_store_document_type: guidance_document_types
+      )
+    end
+
+    it 'filters out content that belongs to a document collection' do
+      assert_includes_params(filter_document_collections: '_MISSING')
+    end
+  end
+
+private
+
+  def assert_includes_params(expected_params)
+    search_results = mock(documents: %w(doc1 doc2))
```

Comment:
I see your point. The reason I stub `RummagerSearch` rather than using webmock is because this is a unit test and I think its boundaries should be described by its immediate interfaces. We could, though, use webmock in our integration tests (https://github.com/alphagov/collections/blob/master/test/integration/taxon_browsing_test.rb#L80-L83 and https://github.com/alphagov/collections/blob/c71b20cb7ca3604239aa9f7f9314da9ad0f90d1a/test/support/rummager_helpers.rb#L2-L15) to make sure the end-to-end system makes sense.

If you agree, what I could do instead is to stub `Services.rummager.search` in the unit tests and assert that it gets the expected parameters (rather than `RummagerSearch`), and in a separate PR work on using webmock in our integration tests.

What do you think?